### ISSUE TITLE
Flag to disable Jenkins CLI

### ISSUE THREAD

**Issue by zloeber** [samrocketman/jenkins-bootstrap-shared#9]
Your bootstrap code and image file automatically disables the jenkins cli. Where would the option be to disable that behavior so I can further run some jenkins job creation scripts (via jenkins-job-builder, a yaml template, and some other scripts I already have.

Thanks for a great project!

**Comment by samrocketman** [comment]
If you're using `jenkins-bootstrap-shared` as a submodule like in the README, then local scripts take precedence over bootstrap-shared scripts when packaging.  Here's an example assuming your repository has a `jenkins-bootstrap-shared submodule:

The shared script is located in:
```bash
$ ls jenkins-bootstrap-shared/scripts/init.groovy.d/disable-jenkins-cli.groovy
```
So you would overwrite it with packaging via

```bash
mkdir -p scripts/init.groovy.d/
touch scripts/init.groovy.d/disable-jenkins-cli.groovy
```

Now, when running `./gradlew clean packages` the empty `disable-jenkins-cli.groovy` overwrites the upstream shared script which means Jenkins CLI will not be disabled.  I would also recommend adding a comment to `scripts/init.groovy.d/disable-jenkins-cli.groovy` so that way other people in your company know the purpose.  For example,

```groovy
/*
   scripts/init.groovy.d/disable-jenkins-cli.groovy overwrites
   jenkins-bootstrap-shared/scripts/init.groovy.d/disable-jenkins-cli.groovy

   This empty script is necessary so that Jenkins CLI is not disabled within
   our packages.  This script overwrites the upstream
   `disable-jenkins-cli.groovy` script.
 */
```

**Comment by samrocketman** [comment]
Also worth noting that you can include your own arbitrary scripts in the `scripts/` directory and any scripts you add to your local `scripts/init.groovy.d/` directory will automatically be included in your own packages.

**Comment by samrocketman** [comment]
In general, I do not manage Jenkins jobs at all and instead expose it to users via self service means.  I recently did a talk (and demo) on this during Jenkins World 2018 this year in San Francisco.  You can see my Jenkins World demo and slides at https://github.com/samrocketman/demo-jenkins-world-2018-jenkins-bootstrap ; check out the diagrams for explanation.

Unfortunately, my Jenkins World presentation was not recorded.  However, around November time I plan to present the same topic for an Online Jenkins Meetup which will be the same presentation I did at Jenkins World (this time recorded).

For creating Jobs, I do not rely on Jenkins CLI because of several security risks and it having a long history of many zero day exploits.  I disable it as a matter of policy in Jenkins instances I manage.  Instead, I create jobs using job XML and the Jenkins REST API.  You can visit any job and append `config.xml` to the URL to see the XML associated with it.

To summarize:

- I only create or two initial jobs using XML and REST APIs (in an idempotent manner).
- The rest of the jobs are created by Jenkins users exposed via self service manner.  See also https://github.com/samrocketman/jenkins-bootstrap-jervis/ which uses my Jervis project https://github.com/samrocketman/jervis/

**Comment by samrocketman** [comment]
Feel free to open other issues or reopen this one.
